src/datasets/arrow_dataset.py

```diff
@@ -5250,6 +5251,8 @@ def push_to_hub(
                 of the logged-in user.
             config_name (`str`, defaults to "default"):
                 The configuration name (or subset) of a dataset. Defaults to "default".
+            as_default_config_name (`bool`, *optional*):
+                Whether to set the config name as default (not necessary if the config_name is "default").
```

Comment:
maybe simpler ?
```suggestion
                Whether to set the config name as default (not necessary if you push only one dataset configuration).
```

Reply:
I know what you mean, but strictly speaking:
- when you call push_to_hub you are pushing just only one configuration; so it is always the case that you push only one
- even if you push multiple configurations (by calling push_to_hub multiple times) and one of them is named "default", it is not necessary either

Reply:
Ah yes correct ! 

https://github.com/huggingface/datasets/blob/2feaa589de86dd85941301fc8c3fa091731a67c0/src/datasets/utils/metadata.py#L235-L235

Maybe this then ?

```suggestion
                Whether to set this configuration as the default one. Otherwise the default configuration is the one named "default"
```